src/app/error.tsx

```diff
@@ -0,0 +1,40 @@
+"use client";
+import React, { useState } from "react";
+import Button from "@mui/material/Button";
+import {
+    ErrorCenterer,
+    ErrorPanel,
+    ErrorLargeText,
+    ErrorSecondaryText,
+    ErrorRetryText,
+} from "@/app/errorPageStyling";
+
+interface ErrorProps {
+    error: Error;
+    reset: () => void;
+}
+
+const ErrorPage: React.FC<ErrorProps> = ({ error, reset }) => {
+    const [ErrorMessage, SetErrorMessage] = useState("");
+
+    const retry = (): void => {
+        SetErrorMessage("Retrying...");
+        setTimeout(() => {
+            reset();
+        }, 1000);
+    };
+    return (
+        <ErrorCenterer>
+            <ErrorPanel elevation={5}>
+                <ErrorLargeText>OOPS!</ErrorLargeText>
+                <ErrorSecondaryText>{error.message}</ErrorSecondaryText>
+                <Button variant="outlined" onClick={retry}>
+                    Try again
+                </Button>
```

Comment:
Assuming this page will be shown every time an error is thrown. In that case would it make more sense for the retry button to be optional for each page? For example, if the add-client fails it is possible that we will have wrong data in the database so we probably don't want them to try again right away?

Reply:
I think we should be able to, as if it is a temporary loss of connection allowing the user to try again will save a user going to a developer!